kubernetes/kubernetes_service.go

```diff
@@ -60,8 +96,24 @@ func (in *IstioClient) GetNamespaces(labelSelector string) ([]core_v1.Namespace,
 // the specified name by querying the cluster API. GetProject will fail
 // if the underlying cluster is not Openshift.
 func (in *IstioClient) GetProject(name string) (*osproject_v1.Project, error) {
-	result := &osproject_v1.Project{}
 
+	if in.k8sCache != nil {
+		projects, err := in.GetProjects("")
```

Comment:
I remember that k8s caches by default had a default index on namespace and name IIRC.
Perhaps this is because we need to create the indexer for projects, right ?